### PR TITLE
CAS-46 Basic postprocessing and visualization capabilities in cellarium-cas

### PR DESCRIPTION
**Pull request by mbabadi**

### CODE REVIEW

File: cellarium/cas/__init__.py [hunk 1]
```diff
@@ -1,4 +1,15 @@
-from cellarium.cas.client import CASClient  # noqa
-from cellarium.cas.version import get_version  # noqa
+from . import constants, exceptions, postprocessing, preprocessing, service, settings, version, visualization
```
Comment (nmalfroy) [inline cellarium/cas/__init__.py:1]: Note that by importing the visualisation in the root package here, we no longer can _not_ install with the `vis` requirements.  Are we ok with that?

**Review by nmalfroy — COMMENTED** [review]
Couple comments so far as I'm working through

File: cellarium/cas/__init__.py [hunk 2]
```diff
@@ -1,4 +1,15 @@
-from cellarium.cas.client import CASClient  # noqa
```
Comment (nmalfroy) [inline cellarium/cas/__init__.py:1]: Note: this breaks the client. We should re-add this line

File: notebooks/quickstart_tutorial.ipynb [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# (This is temporary and will be removed)\n",
+    "import pickle\n",
```
Comment (nmalfroy) [inline notebooks/quickstart_tutorial.ipynb:332]: should this cell still be here?

**Review by nmalfroy — APPROVED** [review]
Approving the bits before my stuff :)

File: cellarium/cas/visualization/__init__.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import logging
+
+try:
+    from .circular_tree_plot_umap_dash_app.app import CASCircularTreePlotUMAPDashApp  # noqa
+    from .ui_utils import find_and_kill_process  # noqa
+
+    __all__ = ["CASCircularTreePlotUMAPDashApp", "find_and_kill_process"]
+except ImportError:
+    logging.debug("Visualization dependencies not installed")
```
Comment (mbabadi) [inline cellarium/cas/visualization/__init__.py:9]: Maybe also say:
"To install Cellarium CAS Client with visualation dependencies from PiPy, please try `pip install --force-reinstall cellarium-cas[vis]`".

File: cellarium/cas/visualization/ui_utils.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    except subprocess.CalledProcessError as e:
+        if verbose:
+            print(f"No process found using port {port}. {e}")
+
+
+T = t.TypeVar("T")
+
+
+class ConfigValue(t.Generic[T]):
```
Comment (mbabadi) [inline cellarium/cas/visualization/ui_utils.py:65]: Nice! :p

File: cellarium/cas/postprocessing/cell_ontology/cell_ontology_cache.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+        # build a networkx graph from CL
+        cl_graph = nx.DiGraph(name="CL graph")
+
+        for cl_class in cl_classes:
+            cl_graph.add_node(cl_class.name)
+
+        for self_cl_class in cl_classes:
+            for parent_cl_class in cl.get_parents_of(self_cl_class):
+                if parent_cl_class not in cl_classes_set:
```
Comment (KevinCLydon) [inline cellarium/cas/postprocessing/cell_ontology/cell_ontology_cache.py:83]: Extremely nitpicky, but would it be very slightly better to do:
```python
if parent_cl_class in cl_classes_set:
    cl_graph.add_edge(parent_cl_class.name, self_cl_class.name)
```

**Review by fedorgrab — APPROVED** [review]

File: cellarium/cas/visualization/circular_tree_plot_umap_dash_app/app.py [hunk 7]
```diff
@@ -0,0 +1,873 @@
...
+from cellarium.cas.postprocessing.cell_ontology import CL_CELL_ROOT_NODE, CellOntologyCache
+from cellarium.cas.visualization._components.circular_tree_plot import CircularTreePlot
+from cellarium.cas.visualization.ui_utils import ConfigValue, find_and_kill_process
+
+# cell type ontology terms (and all descendents) to hide from the visualization
+DEFAULT_HIDDEN_CL_NAMES_SET = {}
+
+# header component ID -> default title mapping for the panels
+DEFAULT_PANEL_TITLES = {
+    "cell-selection-title-tree": "Ontology View",
```
Comment (mbabadi) [inline cellarium/cas/visualization/circular_tree_plot_umap_dash_app/app.py:37]: Cell Type Ontology View

**Review by KevinCLydon — APPROVED** [review]
Only one very minor nitpick that can honestly totally be ignored and it won't really change anything, so 👍

File: cellarium/cas/postprocessing/cell_ontology/cell_ontology_cache.py (hunk 6 above)
Comment (nmalfroy) [reply cellarium/cas/postprocessing/cell_ontology/cell_ontology_cache.py:83]: Hmm...I'll look more closely...would that not reverse the logic?

**Pull request merged**